#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 1c32b9004f8b0f489dc1595ab3fd1edf480cf4ba Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Thu, 4 Jul 2024 15:49:04 +0200
Subject: [PATCH] Draft: Handle cross compiling for tools

This searches qrcAlias and generate-symbolic-dark in
the KF6_HOST_TOOLING path to avoid the complicated
double build structure required to compile them
for the build system when the host differs.
---
 src/tools/CMakeLists.txt | 39 +++++++++++++++++++++++++++++++--------
 1 file changed, 31 insertions(+), 8 deletions(-)

diff --git a/src/tools/CMakeLists.txt b/src/tools/CMakeLists.txt
index 191dfdf3f..952409353 100644
--- a/src/tools/CMakeLists.txt
+++ b/src/tools/CMakeLists.txt
@@ -1,7 +1,30 @@
-# takes light theme icons and makes modified copies of them with dark theme stylesheets
-add_executable(generate-symbolic-dark generate-symbolic-dark.cpp)
-target_link_libraries(generate-symbolic-dark PRIVATE Qt6::Core)
+if (CMAKE_CROSSCOMPILING AND KF6_HOST_TOOLING)
+    find_program(GENERATE_SYMBOLIC_DARK generate-symbolic-dark PATHS ${KF6_HOST_TOOLING}/../../bin NO_CACHE NO_CMAKE_FIND_ROOT_PATH)
+    find_program(QRC_ALIAS qrcAlias PATHS ${KF6_HOST_TOOLING}/../../bin NO_CACHE NO_CMAKE_FIND_ROOT_PATH)
+    if (GENERATE_SYMBOLIC_DARK)
+        add_executable(generate-symbolic-dark IMPORTED GLOBAL)
+        set_target_properties(generate-symbolic-dark PROPERTIES IMPORTED_LOCATION
+                              ${GENERATE_SYMBOLIC_DARK})
+        message (STATUS "Using generate-symbolic-dark from '${GENERATE_SYMBOLIC_DARK}'")
+    else ()
+        message (WARNING "Could not find generate-symbolic-dark in '${KF6_HOST_TOOLING}/../../bin'.")
+    endif ()
+
+    if (QRC_ALIAS)
+        add_executable(qrcAlias IMPORTED GLOBAL)
+        set_target_properties(qrcAlias PROPERTIES IMPORTED_LOCATION
+                              ${QRC_ALIAS})
+        message (STATUS "Using qrcAlias from '${QRC_ALIAS}'")
+    else ()
+        message (WARNING "Could not find qrcAlias in '${KF6_HOST_TOOLING}/bin'.")
+    endif ()
+else ()
+    # takes light theme icons and makes modified copies of them with dark theme stylesheets
+    add_executable(generate-symbolic-dark generate-symbolic-dark.cpp)
+    target_link_libraries(generate-symbolic-dark PRIVATE Qt6::Core)
+
+    # qrc generator
+    add_executable(qrcAlias qrcAlias.cpp)
+    target_link_libraries(qrcAlias PRIVATE Qt6::Core)
+endif ()

-# qrc generator
-add_executable(qrcAlias qrcAlias.cpp)
-target_link_libraries(qrcAlias PRIVATE Qt6::Core)
--
2.45.2
